MapboxCoreNavigation/RouteController.swift

```diff
@@ -611,35 +611,55 @@ extension RouteController: CLLocationManagerDelegate {
 
         for voiceInstruction in spokenInstructions {
             if userSnapToStepDistanceFromManeuver <= voiceInstruction.distanceAlongStep || firstInstructionOnFirstStep {
-
-                NotificationCenter.default.post(name: .routeControllerDidPassSpokenInstructionPoint, object: self, userInfo: [
-                    RouteControllerNotificationUserInfoKey.routeProgressKey: routeProgress
-                ])
-
+                
+                announcePassage(of: voiceInstruction, along: routeProgress)
                 routeProgress.currentLegProgress.currentStepProgress.spokenInstructionIndex += 1
+                
                 return
             }
         }
     }
     
     func updateVisualInstructionProgress() {
         guard let userSnapToStepDistanceFromManeuver = userSnapToStepDistanceFromManeuver else { return }
-        guard let visualInstructions = routeProgress.currentLegProgress.currentStepProgress.remainingVisualInstructions else { return }
+        guard let remainingVisualInstructions = routeProgress.currentLegProgress.currentStepProgress.remainingVisualInstructions else { return }
         
         let firstInstructionOnFirstStep = routeProgress.currentLegProgress.stepIndex == 0 && routeProgress.currentLegProgress.currentStepProgress.visualInstructionIndex == 0
         
-        for visualInstruction in visualInstructions {
+        for visualInstruction in remainingVisualInstructions {
             if userSnapToStepDistanceFromManeuver <= visualInstruction.distanceAlongStep || firstInstructionOnFirstStep {
                 
-                NotificationCenter.default.post(name: .routeControllerDidPassVisualInstructionPoint, object: self, userInfo: [
-                    RouteControllerNotificationUserInfoKey.routeProgressKey: routeProgress
-                    ])
-                
                 routeProgress.currentLegProgress.currentStepProgress.visualInstructionIndex += 1
+                announcePassage(of: visualInstruction, along: routeProgress)
+                
                 return
             }
         }
     }
+    
+    private func announcePassage(of spokenInstructionPoint: SpokenInstruction, along routeProgress: RouteProgress) {
```

Comment:
Agreed, I kind of prefer it this way: "Announce passage of the spoken instruction point along the route's progress"

Reply:
Looks like we normally say “for” a route progress, not “along”. But given that this is a subsequent argument, let’s just leave it unlabeled: `routeProgress`.

Reply:
Ah, I went searching for that pattern but didn’t find it for some reason. 👓

I think we should eliminate that pattern before v1.0 (especially the [labelless `along`](https://github.com/mapbox/mapbox-navigation-ios/blob/dac1496fb758fa8354c527e925113ec9fce96bdf/MapboxCoreNavigation/Router.swift#L21)), because it’s misleading. To reduce the overall churn, let’s start now with just the public methods introduced in this PR; we can deal with the rest of the `along`s separately.